compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -394,6 +394,13 @@ private void visitFunction(BLangFunction funcNode) {
         if (isPublicInvokableNode(funcNode)) {
             analyzeNode(funcNode.returnTypeNode, invokableEnv);
         }
+        for (BLangSimpleVariable requiredParam : funcNode.requiredParams) {
+            analyzeNode(requiredParam, invokableEnv);
+        }
+
+        if (funcNode.restParam != null) {
+            analyzeNode(funcNode.restParam, invokableEnv);
+        }
```

Comment:
In the master, parameters are already validated d68973727541ca5eb8b6cba1a8d424d4bca35140, could you please rebase with master and check. This part will not be needed anymore.